SGA 150 add settings entry to remove all account data

GillWell opened the pull request:
<!-- Enter a brief description/summary of your PR here. What does it fix/what does it change/how was it tested (even manually, if necessary)? -->
## Summary of the Pull Request

<!-- Please review the items on the PR checklist before submitting-->
## PR Checklist

- [x] Delete the authentication data in the authentication page of firebase
- [x] Delete all account documents from all collections
- [x] Delete the user uid from other user friends lists and pending requests

<!-- Provide a more detailed description of the PR, other things fixed or any additional comments/features here -->
## Detailed Description of the Pull Request / Additional comments
Deletes all user data

GillWell committed:
Delete acount

- Added a new setting button for deleting account and user data.

- The button has a confirm dialog

- After confirming the delete documents with the user uid will be deleted.

- The user uid will be deleted from the other users friends lists and pending requests

- It will delete the the user authentication information

GillWell committed:
Account deletion

- Deletes the documents in the collections "inventory", "goals", "journal"

GillWell committed:
Merge branch 'dev' into 150-SGA-Add-settings-entry-to-remove-all-acount-data

GillWell committed:
🤖 Apply Spotless

mantasjasikenas commented:
I get the error when trying to delete my account
![image](https://github.com/PVP-K409/fit-flow/assets/23138747/e2973257-f113-4592-9f97-a589fa728b42)

GillWell committed:
Error fix

- Fixed an error that would happen due to not having done a sign in a long time

- The user has to reauthenticate to be able to delete his account

- If the user chooses a different account the account won't be deleted and a snackbar message will be shown

GillWell committed:
🤖 Apply Spotless

mantasjasikenas commented:
![image](https://github.com/PVP-K409/fit-flow/assets/23138747/5312246c-f89f-48e9-b5b3-d0aa06ea75e9)
